pkg/registrymirror/registrymirrorsstore_test.go

```diff
@@ -0,0 +1,339 @@
+package registrymirror
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	configV1 "github.com/openshift/api/config/v1"
+	operatorV1Alpha1 "github.com/openshift/api/operator/v1alpha1"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	icspA = &operatorV1Alpha1.ImageContentSourcePolicy{
+		ObjectMeta: v1.ObjectMeta{Name: "icspA", UID: "UIDicspA"},
+		Spec: operatorV1Alpha1.ImageContentSourcePolicySpec{
+			RepositoryDigestMirrors: []operatorV1Alpha1.RepositoryDigestMirrors{
+				{Source: "icsp.registry.com", Mirrors: []string{"icsp.mirror1.com", "icsp.mirror2.com"}},
+			},
+		},
+	}
+
+	idmsA = &configV1.ImageDigestMirrorSet{
+		ObjectMeta: v1.ObjectMeta{Name: "idmsA", UID: "UIDidmsA"},
+		Spec: configV1.ImageDigestMirrorSetSpec{
+			ImageDigestMirrors: []configV1.ImageDigestMirrors{
+				{Source: "idms.registry.com", Mirrors: []configV1.ImageMirror{"idms.mirror1.com", "idms.mirror2.com"}},
+			},
+		},
+	}
+
+	itmsA = &configV1.ImageTagMirrorSet{
+		ObjectMeta: v1.ObjectMeta{Name: "itmsA", UID: "UIDitmsA"},
+		Spec: configV1.ImageTagMirrorSetSpec{
+			ImageTagMirrors: []configV1.ImageTagMirrors{
+				{Source: "itms.registry.com", Mirrors: []configV1.ImageMirror{"itms.mirror1.com", "itms.mirror2.com"}},
+			},
+		},
+	}
+)
+
+func fileContains(t *testing.T, path, text string) bool {
+	b, err := os.ReadFile(path)
+	if err != nil {
+		t.Error(err)
+		return false
+	}
```

Comment:
nit: could also be this instead:
```suggestion
require.NoError(t, err)
```